src/qibolab/components/configs.py

```diff
@@ -88,9 +94,48 @@ class AcquisitionConfig:
     iq_angle: Optional[float] = None
     """Signal angle in the IQ-plane for disciminating ground and excited
     states."""
-    kernel: Optional[npt.NDArray] = field(default=None, repr=False)
+    kernel: Annotated[Optional[NdArray], Field(repr=False)] = None
     """Integration weights to be used when post-processing the acquired
     signal."""
 
-
-Config = Union[DcConfig, IqMixerConfig, OscillatorConfig, IqConfig, AcquisitionConfig]
+    def __eq__(self, other) -> bool:
+        """Explicit configuration equality.
+
+        .. note::
+
+            the expliciti definition is required in order to solve the ambiguity about
+            the arrays equality
+        """
+        return (
+            (self.delay == other.delay)
+            and (self.smearing == other.smearing)
+            and (self.threshold == other.threshold)
+            and (self.iq_angle == other.iq_angle)
+            and (self.kernel == other.kernel).all()
+        )
+
+
+class BoundsConfig(Model):
+    """Instument memory limitations proxies."""
+
+    kind: Literal["bounds"] = "bounds"
+
+    waveforms: int
+    """Waveforms estimated size."""
+    readout: int
+    """Number of readouts."""
+    instructions: int
+    """Instructions estimated size."""
+
+
+Config = Annotated[
+    Union[
+        DcConfig,
+        IqMixerConfig,
+        OscillatorConfig,
+        IqConfig,
+        AcquisitionConfig,
+        BoundsConfig,
+    ],
+    Field(discriminator="kind"),
+]
```

Comment:
I generally like the usage of `"kind"` as I believe it simplifies a lot the creation of platforms. However, I am not sure what is the proper way to register additional `Config` kinds here, for instrument specific configs, such as https://github.com/qiboteam/qibolab/blob/0.2/src/qibolab/instruments/qm/components/configs.py. I guess importing here would not work as it would be cyclic.

The temporary solution I found is to use the kind of the parent config and change `model_config` to `extra="allow"`. It is not ideal though.

Reply:
Yes, this is a problem I truly left open: the big deal is not much that you need to specify that `kind` is the discriminator, but rather that we need a single list of all the possible configs when we go for the deserialization.

We don't need to write this list here, so we could even move outside. The only place where this has to be known is in `qibolab.parameters` (and, of course, all its downstreams, e.g. `qibolab.platforms`).

Reply:
If it could solve your problem, moving `Config` to `qibolab.parameters` itself is a valid option.

E.g. we could have a list of `Config` from `qibolab.components`, and join it with a list of `Config` from `qibolab.instruments`. But making the full union only in `qibolab.parameters` (i.e. where is needed).

On the one side, I still hope that we won't need many driver-specific configurations (e.g. only using the general channels). But something we need, and it would be silly not to design support for that (e.g. the emulator needs its own settings)

Reply:
Now this is an issue on its own, #990